Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -939,6 +953,35 @@ class BrowserViewController: UIViewController {
         let ua = newRequest.valueForHTTPHeaderField("User-Agent")
         webView.customUserAgent = ua != UserAgent.defaultUserAgent() ? ua : nil
     }
+
+    private func updateFindInPageVisibility(visible visible: Bool) {
+        if visible {
+            if findInPageBar == nil {
+                let findInPageBar = FindInPageBar()
+                self.findInPageBar = findInPageBar
+                findInPageBar.delegate = self
+                findInPageContainer.addSubview(findInPageBar)
+
+                findInPageBar.snp_makeConstraints { make in
+                    make.edges.equalTo(findInPageContainer)
+                    make.height.equalTo(UIConstants.ToolbarHeight)
+                }
+
+                updateViewConstraints()
+
+                findInPageBar.layoutIfNeeded()
```

Comment:
We actually do need to call `layoutIfNeeded` here because the adding the find in page bar will make it become the first responder, which will in turn fire the `keyboardWillShowWithState` delegate method, and that will animate any new constraints since the last layout pass. That means the constraints set above will be animated in, which we don't want (this causes the bar to slide in from the bottom left corner).

I'll add a comment since this is a pretty subtle detail.